src/libs/auth.js

```diff
@@ -155,6 +151,14 @@ authStore.subscribe(withErrorReporting('Error checking TOS', async (state, oldSt
   }
 }))
 
+authStore.subscribe(async (state, oldState) => {
+  if (!oldState.acceptedTos && state.acceptedTos) {
```

Comment:
Question/TOL: When a user who has already accepted the ToS signs in, do we first write `acceptedTos` as falsy and then truthy? If not, I worry that we might now miss that case. If so, that behavior isn't clear here... which honestly is more or less true of all of these listeners.

I think I just answered my own question by looking at the initialization of `authStore`:
```
export const authStore = Utils.atom({
  isSignedIn: undefined,
  registrationStatus: undefined,
  acceptedTos: undefined,
  user: {},
  profile: {}
})
```

I think I'm starting to feel uncomfortable with how these subscriptions work. I get that it's convenient to be able to watch for arbitrary changes to atoms. However, I worry about how fragile this is over time. I'm afraid it would be too easy to accidentally change code such that a particular state change never happens and not notice that a subscriber is effectively a no-op as a result.

Is there some way we can make this more clear?